python_modules/libraries/dagster-databricks/dagster_databricks/databricks_pyspark_step_launcher.py

```diff
@@ -418,25 +418,23 @@ def _get_step_records() -> Sequence[EventLogEntry]:
             raise
 
     def _grant_permissions(
-        self, log: DagsterLogManager, databricks_run_id: int, request_retries: int = 3
+        self, log: DagsterLogManager, databricks_run_id: int, request_retries: int = 6
     ) -> None:
         client = self.databricks_runner.client.workspace_client
         # Retrieve run info
         cluster_id = None
         for i in range(1, request_retries + 1):
             run_info = client.jobs.get_run(databricks_run_id)
-            if run_info.cluster_instance is None:
-                check.failed("Databricks run {databricks_run_id} has null cluster_instance")
```

Comment:
it can take several moments to initialize a run and the cluster_instance so this check running immediately after the run submission cause the run to fail almost always

![Screenshot 2024-10-04 at 10 42 54 AM](https://github.com/user-attachments/assets/c2ead7ba-307d-4339-af39-287a97ff38c5)